src/server/pfs/s3/s3_test.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+// Tests for the PFS' S3 emulation API. Note that, in calls to
+// `tu.UniqueString`, all lowercase characters are used, unlike in other
+// tests. This is in order to generate repo names that are also valid bucket
+// names. Otherwise minio complains that the bucket name is not valid.
```

Comment:
Do you think this means we should move to being case insensitive with our bucket names? That makes sense to me as a general rule, people shouldn't be using `foo` and `Foo` as different repo names, and it'll remove some awkward cases mapping buckets to repos.

Reply:
I'm not sure. Either way we hit an impedance mismatch - either between PFS and DNS (which most or all object stories rely on) or between PFS and case-insensitive filesystems. My gut is to remain case-sensitive, but don't have a good argument for it.

Reply:
Aren't DNS and the filesystem both case-insensitve? I'm not sure I understand where we'd get into trouble if we went case-insensitive. Either way this isn't a decision that needs to be made for this PR.

Reply:
Depends on the filesystem, but Linux filesystems are almost always case sensitive